tools/Modules/TestFx-Tasks.psm1

```diff
@@ -1,4 +1,160 @@
-﻿[CmdletBinding]
+﻿Function New-AzureCredential
+{
+    [CmdletBinding(
+        DefaultParameterSetName='CreateSpnParamSet', 
+        SupportsShouldProcess=$true
+        )]
+    param(
```

Comment:
I am not super-excited about this idea of using the parameter name prefix to determine whether we are creating the credential.  There aren't any compelling ways to do this, we might consider:

(1) Checking for existence of the SPN, and asking if they want to create it if it does not exist (ShouldContinue)
(2) Adding a switch parameter